### PR TITLE
add tests for missing journal and snapshot plugin config, #24687

### PR DESCRIPTION
**Pull request by patriknw**
Refs #24687

### CODE REVIEW

**Review by chbatey — APPROVED** [review]
LGTM

File: akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedSettings.scala [hunk 1]
```diff
@@ -40,6 +40,8 @@ import com.typesafe.config.Config
     val recoveryEventTimeout: FiniteDuration =
       journalConfig.getDuration("recovery-event-timeout", TimeUnit.MILLISECONDS).millis
 
+    verifySnapshotConfig(config, snapshotPluginId)
```
Comment (helena) [inline akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedSettings.scala]: I don't like these one-off's everywhere, proposing something.

File: akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedSettings.scala [hunk 2]
```diff
@@ -49,12 +51,25 @@ import com.typesafe.config.Config
...
+    if (!config.hasPath(configPath))
+      throw new IllegalArgumentException(s"Journal plugin [$configPath] is not configured.")
     config.getConfig(configPath).withFallback(config.getConfig(Persistence.JournalFallbackConfigPath))
   }
 
+  private def verifySnapshotConfig(config: Config, snapshotPluginId: String): Unit = {
+    if (snapshotPluginId != "" && !config.hasPath(snapshotPluginId))
+      throw new IllegalArgumentException(s"Snapshot store plugin [$snapshotPluginId] is not configured.")
+  }
+
```
Comment (helena) [inline akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedSettings.scala]: Nit that these checks are sprinkled throughout the codebase. WDYT about https://github.com/akka/akka/pull/26559

**Review by johanandren — APPROVED** [review]

File: akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedSettings.scala (hunk 2 above)
Comment (patriknw) [reply akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedSettings.scala]: @helena ok, I extracted some common validation: 5dc23e3

**Review by helena — APPROVED** [review]
LGTM

**Review by johanandren — APPROVED** [review]

**Comment by patriknw** [comment]
rebased, will merge after validation

**Pull request merged**